### ISSUE TITLE
[question] Missing or timed out dynamic request to resource

### ISSUE THREAD

**Issue by wsdookadr** [webrecorder/browsertrix-crawler#664]
If I crawl a website with mostly static resources, I'm noticing there can be missing resources in the resulting WARC. Those are missing either because of broken liks or they timed out during crawling.

I have written tools to grab all WARC-Target-URI and also go through all the Content-Type: text/html WARC records, pull all the urls found in <img src> or <link href> or <a href> or <script src> and compare what's being referenced with what's available in the WARC and bring those in as well. That works fine.

I have the same question but for dynamic pages where the requests are made by js. Are those resource URIs kept somewhere?

Thanks!

**Comment by ikreymer** [comment]
Yes, the crawler should be grabbing all of these resources already, via dynamic behaviors.
The autofetch behavior should pull all img, img srcset, etc... (see: https://github.com/webrecorder/browsertrix-behaviors/blob/main/src/autofetcher.ts#L12) and is enabled by default, so you shouldn't need to do anything with wget

Do you have an example where this is failing? You can increase the timeout for behaviors with `--behaviorTimeout` if the crawler is timing out.

The `urn:pageinfo:<url>` record should record all resources that were requested on the page, including if they failed.

Do you have an example where this is happening? perhaps there's some other issue
